### ISSUE TITLE
[BUG]: x86-64 icc 18.0.0 fails to link

### ISSUE THREAD

**Issue by carenas** [compiler-explorer/compiler-explorer#6654]
### Describe the bug

Probably a missing library from when the compiler was installed, or an incompatible version

### Steps to reproduce

1. visit [godbolt](https://godbolt.org/z/Ejd4qcodK)
2. input any code or flags 
3. Observe the following error in the output window:
```
ASM generation compiler returned: 0
/opt/compiler-explorer/gcc-6.3.0/bin/../lib/gcc/x86_64-linux-gnu/6.3.0/../../../../x86_64-linux-gnu/bin/ld: cannot find -lm
Execution build compiler returned: 1
```

### Expected behavior

Compilation shouldn't show any warnings and `$?` should be 0

### Reproduction link

Not applicable

### Screenshots

Not applicable

### Operating System

NA

### Browser version

NA

**Comment by carenas** [comment]
a similar problem also shows for `x86-64 clang 3.0.0 .. 3.3`, but interestingly all other `icc` versions seem to be working fine.

**Comment by mattgodbolt-molty** [comment]
Closing as won't-fix — icc 18.0.0 is quite ancient and the missing `-lm` is likely a sysroot/library issue from when it was originally installed. Not worth debugging at this point. The same applies to the related clang 3.0–3.3 issue mentioned in the comments.

If this is still important to anyone, please reopen and we can take another look.

*(I'm Molty, an AI assistant acting on behalf of @mattgodbolt)*
